proc-macros/src/helpers.rs

```diff
@@ -193,9 +208,52 @@ pub(crate) fn extract_doc_comments(attrs: &[syn::Attribute]) -> TokenStream2 {
 	quote! ( #(#docs)* )
 }
 
+/// Verify and rewrite the return type.
+///
+/// # Panics
+///
+/// Panics if a `Result/RpcResult` ident is found with wrong arity.
+pub(crate) fn is_result(ty: syn::ReturnType) -> bool {
+	let syn::ReturnType::Type(_, mut ty,) = ty else {
+		return false;
+	};
+
+	// We expect a valid type path.
+	let syn::Type::Path(ref mut type_path) = *ty else  {
+		return false
+	};
+
+	// The path (eg std::result::Result) should have a final segment like 'Result'.
+	let Some(type_name) = type_path.path.segments.last_mut() else {
+		return false
+	};
+
+	// Get the generic args eg the <T, E> in Result<T, E>.
+	let PathArguments::AngleBracketed(AngleBracketedGenericArguments { args, .. }) = &mut type_name.arguments else {
+		return false;
+	};
+
+	if type_name.ident == "Result" {
+		// Result<T, E> should have 2 generic args.
+		if args.len() != 2 {
+			panic!("Result type aliases are not allowed only `std::result::Result<T, E>` and `jsonrpsee::core::RpcResult` are supported");
```

Comment:
nit: Would it be possible to also include the `Span` of the return type in the error?

Maybe returing the `TokenStream` or even panicking with the span info 
```rust
return Err(syn::Error::new(ty.span(), "msg").to_compile_error());
```

We don't necessarily the return types path `std::result::Result<T, E>` maybe renaming to `Result<T, E>` or similar?

Reply:
thanks, I'll revert his and return `Result<(), StirngError` instead I don't like this hacky code

Reply:
it was more to show James what I meant